flax/experimental/nnx/nnx/graph_utils.py

```diff
@@ -903,33 +894,49 @@ def _graph_update_static(
       node_impl.set_key(node, name, value_updates)
 
 
+
 @tp.overload
-def split(graph_node: A) -> tuple[GraphDef[A], State]:
+def full_split(
+  graph_node: A,
+  *,
+  idxmap: dict[Index, tp.Any] | None = None,
+) -> tuple[RefMap[tp.Any, Index], GraphDef[A], State]:
   ...
 
 
 @tp.overload
-def split(
-  graph_node: A, first: filterlib.Filter, /
-) -> tuple[GraphDef[A], State]:
+def full_split(
+  graph_node: A,
+  first: filterlib.Filter,
+  /,
+  *,
+  idxmap: dict[Index, tp.Any] | None = None,
+) -> tuple[RefMap[tp.Any, Index], GraphDef[A], State]:
   ...
 
 
 @tp.overload
-def split(
+def full_split(
   graph_node: A,
   first: filterlib.Filter,
   second: filterlib.Filter,
   /,
   *filters: filterlib.Filter,
-) -> tuple[GraphDef[A], State, tpe.Unpack[tuple[State, ...]]]:
+  idxmap: dict[Index, tp.Any] | None = None,
+) -> tuple[
+  RefMap[tp.Any, Index], GraphDef[A], State, tpe.Unpack[tuple[State, ...]]
+]:
   ...
 
 
-def split(
-  graph_node: A, *filters: filterlib.Filter
-) -> tuple[GraphDef[A], State, tpe.Unpack[tuple[State, ...]]]:
-  graphdef, state, _ = graph_flatten(graph_node)
+def full_split(
```

Comment:
Please ad a docstring for this and the other new APIs.

Reply:
The docstrings are still missing in the latest version. Do you think they are unnecessary and the signatures are sufficient?